Define __setstate__ on _ConvNd to handle pre-padding_mode pickles.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#21687 Define __setstate__ on _ConvNd to handle pre-padding_mode pickles.**

Fixes #20756

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D15807402](https://our.internmc.facebook.com/intern/diff/D15807402)

ezyang committed:
Define __setstate__ on _ConvNd to handle pre-padding_mode pickles.

Fixes #20756

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang commented:
I have a test case but it is blocked on getting access to our AWS bucket so I can upload the test data.

pietern reviewed (approved):
LGTM! If the `none` constant is defined elsewhere (perhaps even as default) then you could use that?

ezyang committed:
Update on "Define __setstate__ on _ConvNd to handle pre-padding_mode pickles."

Define __setstate__ on _ConvNd to handle pre-padding_mode pickles.

Fixes #20756

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 21687 gh/ezyang/167/head

ezyang committed:
Update on "Define __setstate__ on _ConvNd to handle pre-padding_mode pickles."

Define __setstate__ on _ConvNd to handle pre-padding_mode pickles.

Fixes #20756

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 21687 gh/ezyang/167/head

ezyang committed:
Update gitignore for test data on "Define __setstate__ on _ConvNd to handle pre-padding_mode pickles."

Define __setstate__ on _ConvNd to handle pre-padding_mode pickles.

Fixes #20756

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 21687 gh/ezyang/167/head

ezyang committed:
Only pass encoding on Python 3 on "Define __setstate__ on _ConvNd to handle pre-padding_mode pickles."

Define __setstate__ on _ConvNd to handle pre-padding_mode pickles.

Fixes #20756

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 21687 gh/ezyang/167/head

ezyang committed:
Update on "Define __setstate__ on _ConvNd to handle pre-padding_mode pickles."

Define __setstate__ on _ConvNd to handle pre-padding_mode pickles.

Fixes #20756

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

gh-metadata: pytorch pytorch 21687 gh/ezyang/167/head
